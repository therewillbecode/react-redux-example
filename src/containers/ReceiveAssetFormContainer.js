import React, { Component } from "react";
import { connect } from "react-redux";

import { receiveAsset } from "../actions/index";
import ReceiveAssetForm from "../components/ReceiveAssetForm";

export class ReceiveAssetFormContainer extends Component {
  constructor(props) {
    super(props);
    this.state = { name: "", comment: "", nameErr: null };
    this.handleNameChange = this.handleNameChange.bind(this);
    this.handleCommentChange = this.handleCommentChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleNameChange(e, { value }) {
    this.setState({ name: value, nameErr: null });
  }

  handleCommentChange(e, { value }) {
    this.setState({ comment: value });
  }

  handleSubmit() {
    const { name, comment } = this.state;

    if (!name) {
      this.setState({ nameErr: "Name is a required field" });
      return;
    } else {
      this.props.receiveAsset(name, comment);
      this.setState({ name: "", comment: "", nameErr: null });
    }
  }

  render() {
    const { name, comment } = this.state;
    const { nameErr } = this.state;

    return (
      <ReceiveAssetForm
        name={name}
        nameErr={nameErr}
        comment={comment}
        handleCommentChange={this.handleCommentChange}
        handleNameChange={this.handleNameChange}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    receiveAsset: (name, comment) => dispatch(receiveAsset(name, comment))
  };
};

export default connect(undefined, mapDispatchToProps)(
  ReceiveAssetFormContainer
);
