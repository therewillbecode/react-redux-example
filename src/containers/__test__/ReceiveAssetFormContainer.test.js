/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";

import { ReceiveAssetFormContainer } from "../ReceiveAssetFormContainer";

describe("ReceiveAssetFormContainer", () => {
  it("renders correctly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ReceiveAssetFormContainer />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
