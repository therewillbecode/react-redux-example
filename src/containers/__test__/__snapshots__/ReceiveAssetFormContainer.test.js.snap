// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiveAssetFormContainer renders correctly 1`] = `
<ReceiveAssetForm
  comment=""
  handleCommentChange={[Function]}
  handleNameChange={[Function]}
  handleSubmit={[Function]}
  name=""
  nameErr={null}
/>
`;
